### ISSUE TITLE
RCS1118 suggests `const` for a local variable that is passed as an `in` argument

### ISSUE THREAD

**Issue by NoahStolk** [JosefPihrt/Roslynator#1781]
**Product and Version Used**: Roslynator.Analyzers 4.15.0 (also reproduces on current `main`)

**Steps to Reproduce**:

RCS1118 reports a local variable that is later passed to a parameter with an explicit `in` argument keyword. Applying the fix produces code that does not compile, because a constant cannot be passed by readonly reference.

```csharp
class C
{
    void M()
    {
        int repeat = 0; // RCS1118 reported here
        Bar(in repeat);
    }

    void Bar(in int p)
    {
    }
}
```

The same happens when the parameter is `ref readonly` instead of `in`.

I originally hit this with Silk.NET:

```csharp
int repeat = (int)GLEnum.Repeat;
_gl.TexParameterI(TextureTarget.Texture2D, TextureParameterName.TextureWrapS, in repeat);
```

**Actual Behavior**:

RCS1118 is reported, and the code fix produces code that fails to compile with `error CS8156: An expression cannot be used in this context because it may not be passed or returned by reference`:

```csharp
class C
{
    void M()
    {
        const int repeat = 0;
        Bar(in repeat); // error CS8156
    }

    void Bar(in int p)
    {
    }
}
```

**Expected Behavior**:

No diagnostic is reported, since the local variable cannot be marked as const:

```csharp
class C
{
    void M()
    {
        int repeat = 0;
        Bar(in repeat);
    }

    void Bar(in int p)
    {
    }
}
```

---

**Cause** (in case it is useful):

`MarkLocalVariableAsConstWalker` inherits `VisitArgument` from `AssignedExpressionWalker`, which only handles `ref` and `out`, so an `in` argument is not seen as a usage that prevents `const`. `UseConstantInsteadOfFieldWalker` (RCS1187) already has an explicit `SyntaxKind.InKeyword` case for the same situation.

I'd be happy to submit a PR for this. *EDIT: See https://github.com/dotnet/roslynator/pull/1782*

**Note on a related case that is _not_ part of this report**: when a `ref readonly` parameter is called without an argument keyword (`Bar(repeat)`), marking the local as const does not break the build; it only changes warning CS9192 to CS9193. Only an explicit `in`/`ref` at the call site is a hard error.
